kedro-datasets/kedro_datasets/databricks/managed_table_dataset.py

```diff
@@ -236,225 +102,104 @@ def __init__(  # noqa: PLR0913
         """Creates a new instance of ``ManagedTableDataset``.
 
         Args:
-            table: the name of the table
-            catalog: the name of the catalog in Unity.
+            table: The name of the table.
+            catalog: The name of the catalog in Unity.
                 Defaults to None.
-            database: the name of the database.
+            database: The name of the database.
                 (also referred to as schema). Defaults to "default".
-            write_mode: the mode to write the data into the table. If not
+            write_mode: The mode to write the data into the table. If not
                 present, the data set is read-only.
                 Options are:["overwrite", "append", "upsert"].
                 "upsert" mode requires primary_key field to be populated.
                 Defaults to None.
             dataframe_type: "pandas" or "spark" dataframe.
                 Defaults to "spark".
-            primary_key: the primary key of the table.
+            primary_key: The primary key of the table.
                 Can be in the form of a list. Defaults to None.
             version: kedro.io.core.Version instance to load the data.
                 Defaults to None.
-            schema: the schema of the table in JSON form.
+            schema: The schema of the table in JSON form.
                 Dataframes will be truncated to match the schema if provided.
-                Used by the hooks to create the table if the schema is provided
+                Used by the hooks to create the table if the schema is provided.
                 Defaults to None.
-            partition_columns: the columns to use for partitioning the table.
+            partition_columns: The columns to use for partitioning the table.
                 Used by the hooks. Defaults to None.
-            owner_group: if table access control is enabled in your workspace,
+            owner_group: If table access control is enabled in your workspace,
                 specifying owner_group will transfer ownership of the table and database to
                 this owner. All databases should have the same owner_group. Defaults to None.
             metadata: Any arbitrary metadata.
                 This is ignored by Kedro, but may be consumed by users or external plugins.
         Raises:
-            DatasetError: Invalid configuration supplied (through ManagedTable validation)
+            DatasetError: Invalid configuration supplied (through ``ManagedTable`` validation).
         """
-
-        self._table = ManagedTable(
-            database=database,
-            catalog=catalog,
+        super().__init__(
             table=table,
+            catalog=catalog,
+            database=database,
```

Comment:
Can the order be preserved? this would be a breaking change.

Reply:
I just pushed a commit. Does this resolve your concern? Do the order of operations matter here since I am passing in keyword arguments?

Reply:
The dataset class order doesn't matter because it is already a keyword only class.

The manage table class however is technically still a public class so change of order is a breaking change. It's not gonna affect too much since it's rare but I prefer only introduce breaking change when necessary.

Reply:
@noklam Very clear. Thank you.